.github/workflows/daily-tests.yaml

```
@@ -71,141 +62,85 @@ jobs:
       matrix:
-        test-type: [gem5-library-example-x86-ubuntu-run-ALL-x86_64-opt, gem5-library-example-riscv-ubuntu-run-ALL-x86_64-opt, lupv-example-ALL-x86_64-opt, gem5-library-example-arm-ubuntu-run-test-ALL-x86_64-opt, gem5-library-example-riscvmatched-hello-ALL-x86_64-opt]
-    container: gcr.io/gem5-test/ubuntu-22.04_all-dependencies:latest
-    needs: [name-artifacts, build-gem5]
-    timeout-minutes: 1440 # 24 hours
+        suite-id: ${{ needs.testlib-long-get-suites.outputs.suite-ids }}
     steps:
-    - name: Clean runner
-      run:
-        rm -rf ./* || true
-        rm -rf ./.??* || true
-        rm -rf ~/.cache || true
-    - uses: actions/checkout@v3
-      with:
-        # Scheduled workflows run on the default branch by default. We
-        # therefore need to explicitly checkout the develop branch.
-        ref: develop
-    - uses: actions/download-artifact@v3
-      with:
-        name: ${{needs.name-artifacts.outputs.build-name}}ALL
-        path: build/ALL
-    - run: chmod u+x build/ALL/gem5.opt
-    - name: long ${{ matrix.test-type }} gem5_library_example_tests
-      working-directory: ${{ github.workspace }}/tests
-      run: ./main.py run --uid SuiteUID:tests/gem5/gem5_library_example_tests/test_gem5_library_examples.py:test-${{ matrix.test-type }} --length=long --skip-build -vv
-    - name: create zip of results
-      if: success() || failure()
-      run: |
-        apt-get -y install zip
-        zip -r output.zip tests/testing-results
-    - name: upload zip
-      if: success() || failure()
-      uses: actions/upload-artifact@v3
-      env:
-        MY_STEP_VAR: ${{ matrix.test-type }}_COMMIT.${{github.sha}}_RUN.${{github.run_id}}_ATTEMPT.${{github.run_attempt}}
-      with:
-        name: ${{ env.MY_STEP_VAR }}
-        path: output.zip
-        retention-days: 7
-    - run: echo "This job's status is ${{ job.status }}."
+        - name: Clean runner
+          run:
+            rm -rf ./* || true
+            rm -rf ./.??* || true
+            rm -rf ~/.cache || true
+
+        - uses: actions/checkout@v3
+          with:
+            # Scheduled workflows run on the default branch by default. We
+            # therefore need to explicitly checkout the develop branch.
+            ref: develop
+
+          # Here we download ALL the gem5 binaries as we have no way of mapping
+          # suites to the gem5 binaries they actually require, so we obtain all
+          # of them. TODO: This should be fixed. It's an inefficiency.
+        - name: Download gem5 binaries
+          uses: actions/download-artifact@v2
+          with:
+            name: "run-${{ github.run_number }}_attempt-${{ github.run_attempt }}_gem5-builds"
+
+          # GitHub's artifact action does not pereserve file permissions.
+          # Therefore, we need to make the gem5 binaries executable.
+        - name: Chmod gem5.{opt,debug,fast} to be executable
+          run:
+            find . -name "gem5.opt" | xargs chmod u+x
```

nit, personally I'd use the following, but `xargs` works, so no need to change.

```suggestion
            find . -name "gem5.opt" -exec chmod u+x {} \;
```

---

Should be in the latest push.